Microsoft.Toolkit/Diagnostics/ThrowHelper.ThrowExceptions.cs

```diff
@@ -38,33 +34,287 @@ private static string ToAssertString(this object? obj)
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="InvalidOperationException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowInvalidOperationException(string message)
+        {
+            throw new InvalidOperationException(message);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="MissingFieldException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="MissingFieldException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowMissingFieldException(string message)
+        {
+            throw new MissingFieldException(message);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="MissingFieldException"/>.
+        /// </summary>
+        /// <param name="className">The target class being inspected.</param>
+        /// <param name="fieldName">The target field being retrieved.</param>
+        /// <exception cref="MissingFieldException">Thrown with <paramref name="className"/> and <paramref name="fieldName"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowMissingFieldException(string className, string fieldName)
+        {
+            throw new MissingFieldException(className, fieldName);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="MissingMemberException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="MissingMemberException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowMissingMemberException(string message)
+        {
+            throw new MissingMemberException(message);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="MissingMemberException"/>.
+        /// </summary>
+        /// <param name="className">The target class being inspected.</param>
+        /// <param name="memberName">The target member being retrieved.</param>
+        /// <exception cref="MissingMemberException">Thrown with <paramref name="className"/> and <paramref name="memberName"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowMissingMemberException(string className, string memberName)
+        {
+            throw new MissingMemberException(className, memberName);
+        }
+
+        /// <summary>
+        /// Throws a new <see cref="MissingMethodException"/>.
+        /// </summary>
+        /// <param name="message">The message to include in the exception.</param>
+        /// <exception cref="MissingMethodException">Thrown with <paramref name="message"/>.</exception>
+        [DoesNotReturn]
+        public static void ThrowMissingMethodException(string message)
```

Comment:
I don't think providing the `Missingxxx` exceptions is ideal. These probably shouldn't be thrown by a user application, and instead be propagated by the user application when system reflection throws one. I can see a use for them - but they also aren't very performance critical relevant types, and so maybe aren't suitable

Reply:
I mean, I just put them here for completeness, since they're available. I could see cases where users might still want to throw them, eg. when doing reflection on some external types and validating members. Eg. you might query all members and then check them for something, so you would have to throw the exception yourself if you don't find a match.
I agree that when doing reflection performance is not critical, but I'd argue the `ThrowHelper` method would still be useful for code style, clarity and consistency? As in, to allow users to _always_ just use `ThrowHelper` to throw exceptions in the entire codebase.
As opposed to having some `throw new`, some `ThrowHelper` mixed up all over the place.
Thoughts?

Reply:
Sounds reasonable to me - any other exceptions you can think of that might still be useful? Off the top of my head are
```cs
InvalidDataException
LockRecursionException
SynchronizationLockException
Win32Exception
ExternalException
```

Reply:
Ah, yeah good idea! I missed those as I was looking at the list of exceptions from [here](https://www.completecsharptutorial.com/basic/complete-system-exception.php) and those were not mentioned. I'll go ahead and add them too 👍